pkg/arvo/sys/vane/gall.hoon

```diff
@@ -2,1457 +2,2882 @@
+    ::  don't enqueue the subscriber.  Otherwise we increment the meter for
+    ::  the current bone and update the agent's state with it.
+    ::
+    ::  Returns a yes if the meter has been incremented, and no otherwise.
+    ::
+    ++  ap-enqueue
+      ^-  [? _ap-state]
+      ::
+      =/  meter  (~(gut by meter.subscribers.current-agent) agent-bone 0)
+      =/  subscriber=(unit (pair ship path))
+        (~(get by incoming.subscribers.current-agent) agent-bone)
+      ::
+      ?:  ?&  =(20 meter)
+              ?|  ?=(~ subscriber)
+                  !=(our p.u.subscriber)
+              ==
+          ==
+        =/  incoming  (~(get by incoming.subscribers.current-agent) agent-bone)
+        =/  duct  (~(get by duct-map.ducts.current-agent) agent-bone)
+        ~&  [%gall-pulling-20 agent-bone incoming duct]
+        [%.n ap-state]
+      ::
+      =/  next
+        =/  meter
+          (~(put by meter.subscribers.current-agent) agent-bone +(meter))
+        ap-state(meter.subscribers.current-agent meter)
+      ::
+      [%.y next]
+    ::
+    ::  +ap-find-arm: general arm.
+    ::
+    ++  ap-find-arm
+      ~/  %ap-find-arm
+      |=  [=term =path]
+      ^-  [(unit (pair @ud @tas)) _ap-state]
+      ::
+      =/  maybe-cached  (~(get by arm-cache.current-agent) [term path])
+      ?^  maybe-cached
+        [u.maybe-cached ap-state]
+      ::
+      =/  result
+        =/  dep  0
+        |-  ^-  (unit (pair @ud @tas))
+        =/  spu
+          ?~  path
+            ~
+          =/  hyped  (cat 3 term (cat 3 '-' i.path))
+          $(path t.path, dep +(dep), term hyped)
+        ::
+        ?^  spu
+          spu
+        ::
+        ?.  (ap-exists-arm term)
+          ~
+        (some [dep term])
+      ::
+      =.  arm-cache.current-agent
+        (~(put by arm-cache.current-agent) [term path] result)
+      ::
+      [result ap-state]
+    ::
+    ::  +ap-exists-arm: check for an arm in the running agent state.
+    ::
+    ++  ap-exists-arm
+      ~/  %ap-exists-arm
+      |=  =term
+      ^-  ?
+      ::
+      =/  =type  p.running-state.current-agent
+      (slob term type)
+    ::
+    ::  +ap-give: return result.
+    ::
+    ++  ap-give
+      |=  =internal-gift
+      ^+  ap-state
+      ::
+      =/  internal-moves
+        =/  move  [%give internal-gift]
+        =/  =internal-move  [agent-bone move]
+        [internal-move agent-moves]
+      ::
+      ap-state(agent-moves internal-moves)
+    ::
+    ::  +ap-construct-bowl: set up bowl.
+    ::
+    ++  ap-construct-bowl
+      ^+  ap-state
+      ::
+      %_    ap-state
+          +12.q.running-state.current-agent
+        ^-   bowl
+        :*  :*  our                                     ::  host
+                attributing.routes.agent-privilege      ::  guest
+                agent-name                              ::  agent
+            ==                                          ::
+            :*  wex=~                                   ::  outgoing
```

Comment:
@jtobin would tracking/providing these be fairly straightforward for you to implement at this point? If not, can we get a `TODO` marker on this?